### PR TITLE
Fix clang-5 warnings

### PR DESCRIPTION
**Pull request by eustas**
See #28

### CODE REVIEW

**Review by veluca93 — APPROVED** [review]
I'm not sure how well this go but worth a try...

**Comment by deymo** [comment]
It didn't go that far :-/ Maybe try running this workflow on `ubuntu-18.04` (bionic) instead of latest.

**Comment by eustas** [comment]
Yup, trying on ubuntu-18

**Comment by eustas** [comment]
Depends on HWY roll after https://github.com/google/highway/commit/8fa39c099f26265e76a961f5ceaa70427a4d4669

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -17,40 +17,56 @@ on:
...
+            os: ubuntu-latest
             test_in_pr: true
           - name: debug
+            os: ubuntu-latest
           # Build scalar-only hwy instructions.
           - name: scalar
+            os: ubuntu-latest
             mode: release
             cxxflags: -DHWY_DISABLED_TARGETS=~HWY_SCALAR
+          # TODO(eustas): why ASAN has no params?
```
Comment (deymo) [inline .github/workflows/build_test.yml:37]: `mode` is the parameter we pass to ./ci.sh, and if mode is empty we use `name`.

asan, msan and debug don't need any parameters and that's why they are listed above. The rest don't need to be listed afaik.

File: .github/workflows/build_test.yml [hunk 2]
```diff
@@ -17,40 +17,56 @@ on:
 jobs:
   ubuntu_build:
     name: Ubuntu Build ${{ matrix.name }}
-    runs-on: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
```
Comment (deymo) [inline .github/workflows/build_test.yml:20]: `${{ matrix.os || 'ubuntu-latest' }}` and you can remove all the `os: ubuntu-latest` lines.

File: .github/workflows/build_test.yml [hunk 3]
```diff
@@ -17,40 +17,56 @@ on:
...
+            os: ubuntu-latest
             mode: release
             cxxflags: -DJXL_DEBUG_ON_ABORT=0
             cmake_args: >-
               -DJPEGXL_ENABLE_TRANSCODE_JPEG=OFF
               -DJPEGXL_ENABLE_PLUGINS=OFF
               -DJPEGXL_ENABLE_VIEWERS=OFF
           # Builds with gcc in release mode
-          - name: release:gcc8
+          - name: release-gcc8
```
Comment (deymo) [inline .github/workflows/build_test.yml:57]: why the rename? in gitlab we had the compiler as `:compiler`

File: .github/workflows/build_test.yml (hunk 1 above)
Comment (eustas) [reply .github/workflows/build_test.yml]: Got it. Should we remove then `release, msan, scalar` from list and empty `debug` expando?

File: .github/workflows/build_test.yml (hunk 2 above)
Comment (eustas) [reply .github/workflows/build_test.yml]: Done. Thanks

File: .github/workflows/build_test.yml (hunk 3 above)
Comment (eustas) [reply .github/workflows/build_test.yml]: Because ":" can not be part of name in flat list in yml. Rolled back.

**Comment by eustas** [comment]
Rolled HWY dep, fixed clang-5 specific syntax problems (unsupported curly brackets elimination). Now it builds successfully. PTAL @deymo.

**Comment by deymo** [comment]
I think it would be better to have the highway update on a separated PR.
I see this is using 0.14 tag now, but I think there are a few new bugs in 0.14 due to the new SSSE3 target (for example a bug that makes it fail to run on older CPUs with no AES -- this is the fix https://github.com/google/highway/commit/241162b235b788dbfec98a89ae6d08290b0af4c4 or the issue that kCLMUL is aliased with kAVX in that file). I would prefer to update to a version of hwy that doesn't have these regressions.

The other fixes (to .cc and .h) SGTM and we can land those now.

**Comment by eustas** [comment]
Separated code changes from HWY rolling + actions workflow changes.
PTAL

**Review by deymo — APPROVED** [review]

**Pull request merged**